Framework/Algorithms/src/ChangePulsetime.cpp

```diff
@@ -50,8 +54,13 @@ void ChangePulsetime::exec() {
   double timeOffset = getProperty("TimeOffset");
   Progress prog(this, 0.0, 1.0, indexSet.size());
 
-  for (auto i : indexSet) {
-    out_ws->getSpectrum(i).addPulsetime(timeOffset);
+  // This loop is ugly but represents the best way
+  // of parallelising the Workspace access using OpenMP.
+  // In future we may be able to use a range-based loop
+  // with openMP.
```

Comment:
Someone may try to change it to a range-based loop in future. I think the comment lets other developers understand why it is written in canonical form.